### ISSUE TITLE
Empty /opt/OpenvStorage/config/arakoon_cacc.ini

### ISSUE THREAD

**Issue by domsj** [openvstorage/framework#1069]
Our dear friend @JeffreyDevloo had an ovs env with 3 nodes on which the root partition was filled up.
After freeing up space some services failed to start. A small investigation later I noticed that /opt/OpenvStorage/config/arakoon_cacc.ini was empty.

This is probably due to the way this file is refreshed and rewritten. It seems the file is updated in place, which can lead to these sort of situations.

The correct way to update the file would be as follows:
- remove arakoon_cacc.ini.tmp if it exists
- write new contents to arakoon_cacc.ini.tmp
- fsync arakoon_cacc.ini.tmp
- rename (mv) arakoon_cacc.ini.tmp to arakoon_cacc.ini
- (optionally) fsync the parent directory

@dejonghb also suggested reading arakoon_cacc.ini.tmp after it has been fsynced to make sure it contains whatever you expected to be there. Ideally using O_DIRECT flag then to read the data?
(Anyway this is optional, as long as the data is synced to the tempfile before renaming it into place you should be fine.)

**Comment by kinvaris** [comment]
Can this get more priority? This is a very critical issue that can really hurt the ovs cluster.

**Comment by khenderick** [comment]
Seems sth like this should be fairly ok to get a consistent file.

```
def write_consistent(filename, data):
    tmp_filename = '{0}~'.format(filename)
    with open(tmp_filename, 'wb') as f:
        f.write(data)
    os.fsync()
    os.rename(tmp_filename, filename)
```

**Comment by toolslive** [comment]
I think that gives a runtime error: `os.fsync` needs an fd arg, and the fd doesn't live outside the `with` scope.

**Comment by domsj** [comment]
The docs at https://docs.python.org/2/library/os.html also mention first doing a flush...

```
If you’re starting with a Python file object f, first do f.flush(),
 and then do os.fsync(f.fileno()),
 to ensure that all internal buffers associated with f are written to disk.
```

**Comment by khenderick** [comment]
@toolslive, hence the `sth like this`

**Comment by khenderick** [comment]
Fixed by #1168 and openvstorage/alba-asdmanager#160, packaged in openvstorage-2.7.6-rev.4309.4de95cf and openvstorage-sdm-1.6.6-rev.443.472a203

**Comment by JeffreyDevloo** [comment]
### Information
We did not encounter this error anymore on our testing environments.
